components/script/dom/node.rs

```diff
@@ -303,17 +329,18 @@ impl Node {
         child.parent_node.set(None);
         self.children_count.set(self.children_count.get() - 1);
 
-        for node in child.traverse_preorder() {
+        for node in child.traverse_preorder(ShadowIncluding::Yes) {
             // Out-of-document elements never have the descendants flag set.
             node.set_flag(
                 NodeFlags::IS_IN_DOC |
+                    NodeFlags::IS_CONNECTED |
                     NodeFlags::HAS_DIRTY_DESCENDANTS |
                     NodeFlags::HAS_SNAPSHOT |
                     NodeFlags::HANDLED_SNAPSHOT,
                 false,
             );
         }
-        for node in child.traverse_preorder() {
+        for node in child.traverse_preorder(ShadowIncluding::Yes) {
```

Comment:
(we can probably merge the two loops, maybe followup)

Reply:
There's [a comment](https://github.com/servo/servo/pull/22743/files/84e10d7b651c860aeefa95aa98833745eeafd3c9#diff-3a91deb90f65260523e1e9bd52154934R344) explaining why the two loops are required.